File: 053__Maximum_Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        current_max = current_sum = nums[0]

        for item in nums[1:]:

            if current_sum < 0:
                current_sum = item

            else:
                current_sum += item

            if current_sum > current_max:
                current_max = current_sum

        return current_max

        

File: Java/053__Maximum_Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int temp = nums[0];
        int result = nums[0];
        for (int i = 1; i < nums.length; i++) {
            temp = Math.max(temp + nums[i], nums[i]);
            result = Math.max(result, temp);
        }
        return result;
    }
}
